src/server/mod.rs

```diff
@@ -216,14 +220,16 @@ pub fn handle_client(mut stream: TcpStream,
 
 pub fn handle_command(world: &WorldState,
                       game: &mut GameState,
-                      unit_targets: &mut HashMap<UnitId, [f64; 2]>, command: &Command) {
+                      unit_targets: &mut HashMap<UnitId, Vec2>, command: &Command) {
     println!("Did receive command {:?}", command);
     match command {
         &Command::Move(id, move_target) => {
             for player in game.players.iter_mut() {
                 for unit in player.units.iter_mut() {
                     if unit.id == id {
                         println!("Found it :)");
+                        // TODO: This can probably be cleaned up with cgmath
+                        // vector operations.
```

Comment:
There is a  TODO here 😨

Reply:
I think it's fine now, even without fancy vector math. Just a bit cleaned up, and with an explicit clamp function.